--- a/deps/Boost/Boost.cmake	2026-05-12 17:55:17.270642906 -0500
+++ b/deps/Boost/Boost.cmake	2026-05-12 17:55:27.284469660 -0500
@@ -32,11 +32,18 @@
     endif ()
 endif ()
 
+if (APPLE)
+    set(_boost_patch_cmd PATCH_COMMAND git apply --verbose --ignore-space-change --whitespace=fix
+        ${CMAKE_CURRENT_LIST_DIR}/0001-FIX-OBS-cannot-start-streaming-on-MAC.patch)
+else ()
+    set(_boost_patch_cmd "")
+endif ()
+
 bambustudio_add_cmake_project(Boost
     URL "https://github.com/boostorg/boost/releases/download/boost-1.84.0/boost-1.84.0.tar.gz"
     URL_HASH SHA256=4d27e9efed0f6f152dc28db6430b9d3dfb40c0345da7342eaa5a987dde57bd95
     LIST_SEPARATOR |
-    PATCH_COMMAND git apply --verbose --ignore-space-change --whitespace=fix ${CMAKE_CURRENT_LIST_DIR}/0001-FIX-OBS-cannot-start-streaming-on-MAC.patch
+    ${_boost_patch_cmd}
     CMAKE_ARGS
         -DBOOST_EXCLUDE_LIBRARIES:STRING=contract|fiber|numpy|wave|test
         -DBOOST_LOCALE_ENABLE_ICU:BOOL=OFF # do not link to libicu, breaks compatibility between distros
